snuba/clickhouse.py

```diff
@@ -218,6 +218,52 @@ def for_schema(self):
         return u'Nullable({})'.format(self.inner_type.for_schema())
 
 
+class Materialized(ColumnType):
+    def __init__(self, inner_type, expression):
+        self.inner_type = inner_type
+        self.expression = expression
+
+    def __repr__(self):
+        return u'Materialized({}, {})'.format(
+            repr(self.inner_type),
+            self.expression,
+        )
+
+    def __eq__(self, other):
+        return self.__class__ == other.__class__ \
+            and self.expression == other.expression \
+            and self.inner_type == other.inner_type
+
+    def for_schema(self):
+        return u'{} MATERIALIZED {}'.format(
+            self.inner_type.for_schema(),
+            self.expression,
+        )
+
+
+class WithDefault(ColumnType):
```

Comment:
I think we can refactor the logic of these column types with inner types to reduce copy paste.

Reply:
Switching these to dataclasses would probably remove half of the code in this file since `__eq__` and `__repr__` would come for free, all of the other required methods are implementation-specific anyway.

Reply:
(To be clear, this is not a change request for this PR, but a suggested followup.)